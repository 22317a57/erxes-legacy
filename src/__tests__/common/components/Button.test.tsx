import { mount, shallow } from 'enzyme';
import React from 'react';

import Button from '../../../modules/common/components/Button';

describe('Button component', () => {
  const defaultProps = {
    btnStyle: 'default',
    size: 'medium',
    block: false,
    type: 'button'
  };

  test('renders successfully', () => {
    shallow(<Button />);
  });

  test('fully renders with default props', () => {
    const rendered = mount(<Button {...defaultProps} />);
    const props = rendered.props();

    expect(defaultProps).toMatchObject(props);
  });

  test('fully renders with custom props', () => {
    const p = {
      ...defaultProps,
      size: 'large',
      type: 'submit',
      block: true
    };

    const rendered = mount(<Button {...p} />);
    const props = rendered.props();

    expect(p).toMatchObject(props);
  });
});
