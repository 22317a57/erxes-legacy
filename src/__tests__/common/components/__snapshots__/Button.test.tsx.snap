// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component snapshot matches 1`] = `
<button
  className="sc-bZQynM jGbjye"
  type="button"
/>
`;
